src/main/java/org/mule/extension/http/internal/listener/profiling/tracing/HttpListenerCurrentSpanCustomizer.java

```diff
@@ -110,4 +121,20 @@ protected String getSpanKind() {
   protected String getSpanName() {
     return attributes.getListenerPath();
   }
+
+  /**
+   * W-12558102
+   * This provides a transparent way to obtain the definitive list of headers to add in the spans associated with Otel tracing.
+   * This list will have all headers except those that have been skipped via the skipHeadersOnTracing property.
+   */
+  @Override
+  public MultiMap<String, String> getHeaders() {
```

Comment:
This method seems to be the same for the HTTP Listener and the requester, could its definition be moved to HttpCurrentSpanCustomizer?